### PR TITLE
added inline documentation and edited some codes

### PR DESCRIPTION
**Pull request by satyamroy001**
1.Inline Documentation 
2. added .cpu() 
why?? reason- The reason for requiring explicit .cpu() is that CPU tensors and the converted numpy arrays share memory. If a .cpu() is implicitly done, the operation will be different for CUDA and CPU tensors, and we wanted to be explicit to avoid bugs.
Reference- https://discuss.pytorch.org/t/should-it-really-be-necessary-to-do-var-detach-cpu-numpy/35489/5

## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Relevant Issue
Issue #Tag the issue here

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by divyanshugit** [comment]
Thanks for the PR @satyamroy001. Can you please go through the [Contributing Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md)

Specifically 
- pep8 [guidelines](https://www.python.org/dev/peps/pep-0008/)
- numpy style [guide](https://numpydoc.readthedocs.io/en/latest/format.html)

File: fedrec/utilities/cuda_utils.py [hunk 1]
```diff
@@ -1,7 +1,24 @@
...
+""" 
+importing necessary (libraries) 
+torch ---> PyTorch is a Python package that provides two high-level features:
+Tensor computation (like NumPy) with strong GPU acceleration
+Deep neural networks built on a tape-based autograd system
+
+socket ---> Python package which allows creation of simple servers and clients for communication with sockets. Supports both Python2 and Python3.
+logging ---> This module is intended to provide a standard error logging mechanism in Python as per PEP 282.
+
+"""
```
Comment (divyanshugit) [inline fedrec/utilities/cuda_utils.py:11]: It will be great. If you can explain the purpose rather than the packages.

File: fedrec/utilities/cuda_utils.py [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
 import socket
 import logging
-
+"""
+function name ---> map_to_cuda 
+if the input to the function is of type "List" or "Tuple" it will return [map_to_cuda(arg, device, **kwargs) for arg in args]
+if the input to the function is of type "Dictionary" it will return {k: map_to_cuda(v, device, **kwargs) for k, v in args.items()}
+if the input to the function is of type "torch.Tensor" it will return args.cuda(device, **kwargs)
+if there is unmatched of all it will raise TypeError "unsupported type for cuda migration"
+"""
```
Comment (divyanshugit) [inline fedrec/utilities/cuda_utils.py:21]: Please refactor it using Numpy style guidelines

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Comment by divyanshugit** [comment]
@satyamroy001 Can you please refactor the inline comments according to Numpy Style guidelines!!

Thanks

File: fedrec/utilities/cuda_utils.py (hunk 1 above)
Comment (satyamroy001) [reply fedrec/utilities/cuda_utils.py:11]: Sure sir @divyanshugit I am working on it today

File: fedrec/utilities/cuda_utils.py (hunk 2 above)
Comment (satyamroy001) [reply fedrec/utilities/cuda_utils.py:21]: Sure sir, I am working on it already completed going to open pull request soon!!!

**Comment by satyamroy001** [comment]
@divyanshugit sir, added Numpy guidelines please have a look? and I need one suggestion do I need to add extra single lines comments to describe the function tasks? if yes then I will improve it a bit.

File: fedrec/utilities/cuda_utils.py [hunk 3]
```diff
@@ -1,9 +1,55 @@
...
+3. torch.cuda is used to set up and run CUDA operations
+
+socket
+1. To get host name of the GPU in mapping processes
+
+logging
+1.emitting log messages from Python programs
+2.Used to confirm that things are working as expected
+3.logging.info(msg) : This will log a message with level INFO on this logger.   
+"""
```
Comment (divyanshugit) [inline fedrec/utilities/cuda_utils.py:19]: As I said earlier, you don't need to explain the libraries/packages separately. Instead of this, you can write a paragraph where you can mention the purpose of the file using the libraries.

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Review by divyanshugit — CHANGES_REQUESTED** [review]

File: fedrec/utilities/cuda_utils.py [hunk 4]
```diff
@@ -13,14 +59,34 @@ def map_to_cuda(args, device=None, **kwargs):
     else:
         raise TypeError("unsupported type for cuda migration")
 
-
 def map_to_list(model_params):
+"""
+def map_to_list(model_params):
```
Comment (divyanshugit) [inline fedrec/utilities/cuda_utils.py:64]: Please take a reference from other files.

File: fedrec/utilities/cuda_utils.py [hunk 5]
```diff
@@ -13,14 +59,34 @@ def map_to_cuda(args, device=None, **kwargs):
...
+"""
     for k in model_params.keys():
-        model_params[k] = model_params[k].detach().numpy().tolist()
+        model_params[k] = model_params[k].detach().cpu().numpy().tolist()
     return model_params
 
-
 def mapping_processes_to_gpus(gpu_config, process_id, worker_number):
+"""
+def mapping_processes_to_gpus(gpu_config, process_id, worker_number):
```
Comment (divyanshugit) [inline fedrec/utilities/cuda_utils.py:79]: Please take a reference from other files.
